pkg/security/tls.go

```diff
@@ -136,9 +132,56 @@ func newClientTLSConfig(certPEM, keyPEM, caPEM []byte) (*tls.Config, error) {
 		return nil, errors.Errorf("failed to parse PEM data to pool")
 	}
 
+	cfg := newBaseTLSConfig()
+	cfg.Certificates = []tls.Certificate{cert}
+	cfg.RootCAs = certPool
+	return cfg, nil
+}
+
+// newBaseTLSConfig returns a tls.Config initialized with the
+// parameters that are common to clients and servers.
+func newBaseTLSConfig() *tls.Config {
 	return &tls.Config{
-		Certificates: []tls.Certificate{cert},
-		RootCAs:      certPool,
-		MinVersion:   tls.VersionTLS12,
-	}, nil
+		MinVersion: tls.VersionTLS12,
+
+		// This is Go's default list of cipher suites (as of go 1.8.3),
+		// with the following differences:
+		// - 3DES-based cipher suites have been removed. This cipher is
+		//   vulnerable to the Sweet32 attack and is sometimes reported by
+		//   security scanners. (This is arguably a false positive since
+		//   it will never be selected: Any TLS1.2 implementation MUST
+		//   include at least one cipher higher in the priority list, but
+		//   there's also no reason to keep it around)
+		// - AES is always prioritized over ChaCha20. Go makes this decision
+		//   by default based on the presence or absence of hardware AES
+		//   acceleration.
+		//   TODO(bdarnell): do the same detection here. See
+		//   https://github.com/golang/go/issues/21167
+		//
+		// Note that some TLS cipher suite guidance (such as Mozilla's[1])
+		// recommend replacing the CBC_SHA suites below with CBC_SHA384 or
+		// CBC_SHA256 variants. We do not do this because Go does not
+		// currerntly implement the CBC_SHA384 suites, and its CBC_SHA256
+		// implementation is vulnerable to the Lucky13 attack and is disabled
+		// by default.[2]
```

Comment:
It's unclear. Mozilla's ["modern compatibility" recommendation](https://wiki.mozilla.org/Security/Server_Side_TLS#Modern_compatibility), which also has a minimum of TLS 1.2, doesn't say why it still includes CBC-based cipher suites. It looks like [Safari](https://www.ssllabs.com/ssltest/viewClient.html?name=Safari&version=8&platform=iOS%208.4&key=85) may have been the last major browser to get GCM support (in version 9). 

We can probably get away with being more aggressive than mozilla's recommendations in removing old cipher suites, but I'm reluctant to take the lead and remove these ciphers without a better understanding of the compatibility implications.